### PR TITLE
Configure whether to show recommendations of external installations

### LINKED ISSUE #1816: Configure whether to show recommendations of external installations
When opening a pom.xml file there is a pop up showing recommendation of dependency analytics:
image

related PR: https://github.com/redhat-developer/vscode-java/pull/1771/files

Is it possible to add configuration in settings.json so it won't show such recommendations at all? 
We know when choosing "Never" the choice is saved in the extension's global state but we don't want to show this info message at all in the first place. 

The reason is that in our landscape we have predefined/close set of installed extensions and not all external extensions can work out of the box. 

##### Environment
- Operating System:
- JDK version: 11
- Visual Studio Code version: 
- Java extension version:

##### Steps To Reproduce
1. Open pom.xml file and see recommendation notification to install dependency analytics

### PR DESCRIPTION
**Pull request by snjeza**
Fixes #1816 

You can use
```
"java.dependency.analytics.show": false
```

Signed-off-by: Snjezana Peco <snjezana.peco@redhat.com>

### CODE REVIEW

File: src/recommendation/dependencyAnalytics.ts [hunk 1]
```diff
@@ -8,12 +8,17 @@ import { IHandler } from "./handler";
 const EXTENSION_NAME = "redhat.fabric8-analytics";
 const GH_ORG_URL = `https://github.com/fabric8-analytics`;
 const RECOMMENDATION_MESSAGE = `Do you want to install the [Dependency Analytics](${GH_ORG_URL}) extension to stay informed about vulnerable dependencies in pom.xml files?`;
+const JAVA_DEPENDENCY_ANALYTICS_SHOW = "java.dependency.analytics.show"
```
Comment (rgrunber) [inline src/recommendation/dependencyAnalytics.ts:11]: ;

Comment (snjeza) [reply src/recommendation/dependencyAnalytics.ts]: Fixed

Comment (rgrunber) [reply src/recommendation/dependencyAnalytics.ts]: (though it build just fine for me)

**Comment by rgrunber** [comment]
:+1: This works for me. It won't respond to changes in the setting after activation (so if you set it to false in settings but don't restart you'll still get the prompts) but I guess we don't need to go that far in this case.

@fbricon any issues with this option ?

File: src/recommendation/dependencyAnalytics.ts (hunk 1 above)
Comment (fbricon) [reply src/recommendation/dependencyAnalytics.ts]: i'd prefer java.recommendations.dependency.analytics.show

**Comment by fbricon** [comment]
workflow is fine

File: src/recommendation/dependencyAnalytics.ts (hunk 1 above)
Comment (snjeza) [reply src/recommendation/dependencyAnalytics.ts]: Fixed.

**Comment by veredcon** [comment]
Thanks for this enhancement!

**Pull request merged**